File: src/main/java/com/aryan/ziplink/service/DashboardService.java
package com.aryan.ziplink.service;

import com.aryan.ziplink.dto.response.DashboardSummaryResponse;
import com.aryan.ziplink.dto.response.UrlResponse;
import com.aryan.ziplink.enums.UrlFilterStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface DashboardService {
    DashboardSummaryResponse getDashboardSummary();
//    Page<UrlResponse> getUrls(String search, UrlFilterStatus status, Pageable pageable);
//    List<UrlResponse> getExpiringSoon(int days);
//    List<UrlResponse> getRecentUrls(int limit);
//    Page<UrlResponse> getTopUrls(Pageable pageable);
}

File: ziplink-front/src/services/dashboard.service.js
import api from "./api"

const dashboardService={
    async getDashboardData(){
        const response=await api.get("/dashboard");
        return response.data.data
    },

    async getUrls(params={}){
        const response =await api.get("/dashboard/urls",{params,});
        return response.data.data;
    },

    async getRecentUrls(limit = 10) {
        const response = await api.get("/dashboard/recent", {
            params: { limit },
        });

        return response.data.data;
    },
    async getTopUrls(params = {}) {
        const response = await api.get("/dashboard/top", {
            params,
        });

        return response.data.data;
    },

    async getExpiringSoon(days = 7) {
        const response = await api.get("/dashboard/expiring", {
            params: { days },
        });

        return response.data.data;
    },

    async getUrlAnalytics(id) {
        const response = await api.get(`/dashboard/urls/${id}/analytics`);

        return response.data.data;
    },
};

export default dashboardService;